File: tests/fixtures/discrimination/python/out_of_corpus/generic_15.py
class SimpleQueue:
    def __init__(self):
        self.items = []

    def enqueue(self, item):
        self.items.append(item)

    def dequeue(self):
        if len(self.items) == 0:
            return None
        return self.items.pop(0)

    def front(self):
        if len(self.items) == 0:
            return None
        return self.items[0]

    def is_empty(self):
        return len(self.items) == 0

    def size(self):
        return len(self.items)


def rotate_list(lst, k):
    if len(lst) == 0:
        return lst
    k = k % len(lst)
    return lst[k:] + lst[:k]


def flatten_list(nested):
    result = []
    for item in nested:
        if isinstance(item, list):
            result.extend(flatten_list(item))
        else:
            result.append(item)
    return result

File: tests/fixtures/discrimination/go/out_of_corpus/generic_15.go
package sample

type IntQueue struct {
	items []int
}

func NewIntQueue() *IntQueue {
	return &IntQueue{items: []int{}}
}

func (q *IntQueue) Enqueue(val int) {
	q.items = append(q.items, val)
}

func (q *IntQueue) Dequeue() (int, bool) {
	if len(q.items) == 0 {
		return 0, false
	}
	val := q.items[0]
	q.items = q.items[1:]
	return val, true
}

func (q *IntQueue) Front() (int, bool) {
	if len(q.items) == 0 {
		return 0, false
	}
	return q.items[0], true
}

func (q *IntQueue) IsEmpty() bool {
	return len(q.items) == 0
}

func (q *IntQueue) Size() int {
	return len(q.items)
}

func (q *IntQueue) ToSlice() []int {
	result := make([]int, len(q.items))
	copy(result, q.items)
	return result
}
